http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/package.scala

```diff
@@ -82,4 +85,39 @@ package object service {
         )
       )
 
+  /**
+   * Parses requested creator email from a query. Returns the calling user's WorkbenchEmail given param role=creator or
+   * creator=(caller's email). The creator=email syntax is provided to support legacy terra-ui usages.
+   * @param userInfo ID and credentials of the calling user
+   * @param params Map of query params
+   * @return Either a ParseCreatorOnlyException or an Option WorkbenchEmail
+   */
+  private[service] def processCreatorOnlyParameter(
+    userInfo: UserInfo,
+    params: LabelMap
+  ): Either[ParseCreatorOnlyException, Option[WorkbenchEmail]] =
+    for {
+      result <- params match {
+        case map if map.isDefinedAt(creatorOnlyKey) =>
+          if (map.get(creatorOnlyKey).contains(creatorOnlyValue))
+            Either.right(Some(userInfo.userEmail))
+          else
+            Either.left(
+              ParseCreatorOnlyException(
+                s"Failed to process invalid value for ${creatorOnlyKey}. The only currently supported value is ${creatorOnlyValue}."
+              )
+            )
+        case map if map.isDefinedAt(creatorOnlyValue) =>
+          if (map.get(creatorOnlyValue).exists(_ == userInfo.userEmail.toString))
+            Either.right(Some(userInfo.userEmail))
+          else
+            Either.left(
+              ParseCreatorOnlyException(
+                s"Failed to process invalid value for ${creatorOnlyValue}. The only currently supported value is your own user email."
+              )
+            )
+        case _ =>
+          Either.right(None)
```

Comment:
this seems a lot more complicated than https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala#L319-L326

can u add some unit tests for this helper...also we can replace https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterp.scala#L319-L326 with the function as well

Not sure if we need to specific `ParseCreatorOnlyException` either, can u just reuse `BadRequestException`..Also populate the `traceId` properly

Reply:
It's definitely more complex to read! I wanted to make sure that the method handled errors similarly to the others in `package.scala`, and also that the errors were relevant to the specific param passed (`role=` vs `creator=`). These can definitely change to `BadRequestException` (I was using the examples of `IllegalLabelKeyException` etc) - will do!

I plan to refactor the listRuntimes methods in a separate PR, per a convo with Liz elsewhere on this page.

Will also add unit tests and the traceId~

Reply:
If there is a simpler way to handle the Either() return structure please let me know; I'd much prefer a cleaner implementation if that's possible

Reply:
>I plan to refactor the listRuntimes methods in a separate PR, per a convo with Liz elsewhere on this page.
I'd prefer we do that in this PR since it's not that big anyway...Otherwise I'd like to see a ticket number for the follow up PR or an actual PR already :) just so I don't have to remember this

>I'd much prefer a cleaner implementation if that's possible
I think what u have is fine...but 
I'd probably do sth like 
```
    val creatorOnlyFromRoleKey = params.get(creatorOnlyKey) match {
        case Some(value) => Either.cond(value == creatorOnlyValue, Some(callerEmail), BadRequestException("???", Some(traceId))) 
        case None => none.asRight[BadRequestException]
      } 
    
    // For backwards compatibility. xxxx
    val creatorOnlyFromCreatorKey = params.get(creatorOnlyValue) match {
        case Some(value) => Either.cond(value == callerEmail, Some(callerEmail), BadRequestException("???", Some(traceId))) 
        case None => none.asRight[BadRequestException]
      }
    
    creatorOnlyFromRoleKey orElse creatorOnlyFromRoleKey
```

Reply:
Magnificent thank you!!